ScienceJournal/UI/TrialDetailViewController.swift

```diff
@@ -1758,9 +1758,40 @@ class TrialDetailViewController: MaterialHeaderViewController,
   }
 
   func cameraButtonPressed() {
-    present(cameraImageProvider.cameraViewController, animated: true, completion: nil)
-  }
+    switch CaptureSessionInterruptionObserver.shared.cameraAvailability {
+    case .permissionsDenied:
+      showCameraPermissionsDeniedAlert()
+    case .blockedByBrightnessSensor:
+      showSnackbar(withMessage: String.inputCameraBlockedByBrightnessSensor,
+                   category: nil,
+                   actionTitle: String.actionOk,
+                   actionHandler: nil
+      )
+    default: // Handles .available and .captureInterrupted
+      present(cameraImageProvider.cameraViewController, animated: true)
+    }
```

Comment:
Is there a way to move the error handling to `CameraImageProvider`, so it could handle both success and failure in order to remove this duplication? I'm not sure how best to handle the `blockedByBrightnessSensor`/snackbar case, but there could be some kind of presentation method that would take the presenting VC as a param and then present either the alert or the camera on it. Perhaps a simple protocol with the `showSnackbar` method that these presenting VCs could conform to, then the image provider could handle all cases.